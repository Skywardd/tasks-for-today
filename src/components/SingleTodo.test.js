
import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import SingleTodo from './SingleTodo';
import '@testing-library/jest-dom'
import { create } from 'domain';

afterEach(()=> {
    cleanup();
})

test('should render SingleTodo not done component', ()=>{  
    let todo ={
        id:2,
        todo:"make wafles",
        isDone: false
    }
    render(<SingleTodo  todo ={todo}
        key={todo.id}
      />);
    const formElement= screen.getByTestId("render-singleTodo" ); 
    expect(formElement).toBeInTheDocument();

  
});

test('should render SingleTodo done component', ()=>{  
    let todo ={
        id:2,
        todo:"make wafles",
        isDone: true
    }
    render(<SingleTodo  todo ={todo}
        key={todo.id}
      />);
    const formElement= screen.getByTestId("render-singleTodo" ); 
    expect(formElement).toBeInTheDocument();

});

test('matches snapshot',()=>{
    let todo ={
        id:2,
        todo:"make wafles",
        isDone: false
    }

    const tree= renderer.create(<SingleTodo  todo ={todo} key={todo.id}/>).toJSON();

    expect(tree).toMatchSnapshot();
})