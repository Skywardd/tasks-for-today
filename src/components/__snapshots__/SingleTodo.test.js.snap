// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="todos__single"
  data-testid="render-singleTodo"
  onSubmit={[Function]}
>
  <span
    className="todos__single--text"
  >
    make wafles
  </span>
  <div>
    <span
      className="icon"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
        />
      </svg>
    </span>
    <span
      className="icon"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
        />
      </svg>
    </span>
    <span
      className="icon"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
        />
      </svg>
    </span>
  </div>
</form>
`;
